frontend/src/components/resourceMap/search/ResourceSearch.tsx

```diff
@@ -0,0 +1,98 @@
+import { Autocomplete, Box, TextField } from '@mui/material';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { KubeObject } from '../../../lib/k8s/cluster';
+import { KubeIcon } from '../kubeIcon/KubeIcon';
+
+const MAX_RESULTS = 8;
+export function ResourceSearch({
+  resources,
+  onSearch,
+}: {
+  resources: KubeObject[];
+  onSearch: (resource: KubeObject) => void;
+}) {
+  const { t } = useTranslation();
+  const [query, setQuery] = useState('');
+
+  const results = useMemo(() => {
+    if (!resources || !query.trim()) return [];
+    const results = [];
+
+    for (let i = 0; i < resources.length; i++) {
+      const resource = resources[i];
+      if (resource.metadata.name.includes(query)) {
+        results.push(resource);
+      }
+      if (results.length >= MAX_RESULTS) {
+        break;
+      }
+    }
+
+    return results;
+  }, [query, resources]);
+
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+      }}
+    >
+      <Autocomplete
+        renderInput={params => (
+          <TextField {...params} label={t('Search')} sx={{ width: '300px' }} size="small" />
+        )}
+        freeSolo
+        clearOnBlur
+        filterOptions={x => x}
+        getOptionLabel={option => (typeof option === 'string' ? option : option.metadata.name)}
+        onInputChange={(e, value) => {
+          setQuery(value);
+        }}
+        onChange={(e, value) => {
+          if (value && typeof value !== 'string') {
+            onSearch(value);
+          }
+        }}
+        options={results}
+        renderOption={(props, it) => {
+          return (
+            <li {...props}>
```

Comment:
html

Reply:
I don't understand, even in the mui example they use li. It's a perfectly reasonable element to use here
https://mui.com/material-ui/react-autocomplete/#search-as-you-type

Reply:
nit, feel free to ignore

Reply:
let's skip this